### PR TITLE
Extend unique index for rhnpackagechangelogdata table

### PR DESCRIPTION
**Pull request by witekest**
## What does this PR change?

In spite of client code trying to avoid duplicate records for changelog
data such records are observed in the database.
The change enforces unique combination of `name`, `time` and `text`
fields by updating the definition of the unique index on the database
level.

Additionally:

- Add logging package details during failed import.
- Clean up the logic for removing duplicate changelog entries.

## Links

Tracks suse/spacewalk#22470

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: python/spacewalk/spacewalk-backend.changes.witek.extend-reposync-logging [hunk 1]
```diff
@@ -0,0 +1 @@
+- Extend reposync logging about incorect changelog entries
```
Comment (mackdk) [inline python/spacewalk/spacewalk-backend.changes.witek.extend-reposync-logging:1]:
```suggestion
- Extend reposync logging about incorrect changelog entries
```

**Review by mackdk — APPROVED** [review]

File: python/spacewalk/satellite_tools/reposync.py [hunk 2]
```diff
@@ -1366,9 +1366,9 @@ def import_package_batch(self, to_process, to_disassociate, is_non_local_repo, b
                     raise
                 except Exception as e:
                     e_message = f'Exception: {e}'
-                    if importer:
+                    if 'importer' in locals():
                         e_message += f'\nPackage: {repr(importer)}'
-                    if src_importer:
+                    if 'src_importer' in locals():
```
Comment (agraul) [inline python/spacewalk/satellite_tools/reposync.py:1371]: Why is this change needed?

**Review by agraul — COMMENTED** [review]
The extra logging looks good to me. I just wonder why the other change is needed.

File: python/spacewalk/satellite_tools/reposync.py (hunk 2 above)
Comment (witekest) [reply python/spacewalk/satellite_tools/reposync.py:1371]: One of the variables `importer` or `src_importer` is not assigned at that point and the `UnboundLocalError` is thrown.

Anyway, `repr(importer)` does not provide useful information, so I'll drop that part.

**Comment by witekest** [comment]
@cbosdo Could you take a look at the sonarcloud scanner failure?

**Comment by cbosdo** [comment]
> @cbosdo Could you take a look at the sonarcloud scanner failure?

It looks like PRs coming form forks don't have access to the sonar token...

**Comment by witekest** [comment]
@mackdk Could you please review the suggested new unique index for changelog entries? Thanks in advance.

**Review by mackdk — COMMENTED** [review]
Looks good, but also file `schema/spacewalk/common/tables/rhnPackageChangeLogData.sql` should be updated to reflect the new index.

File: schema/spacewalk/common/tables/rhnPackageChangeLogData.sql [hunk 3]
```diff
@@ -28,8 +28,9 @@ CREATE TABLE rhnPackageChangeLogData
 
 ;
 
-CREATE INDEX rhn_pkg_cld_nt_idx
-    ON rhnPackageChangeLogData (name, time)
+CREATE UNIQUE INDEX CONCURRENTLY rhn_pkg_cld_ntt_idx
```
Comment (mackdk) [inline schema/spacewalk/common/tables/rhnPackageChangeLogData.sql:31]: I don't think `CONCURRENTLY` is accepted in a `tables` file because, as far as I know, these scripts are used to create the database from scratch.

Comment (witekest) [reply schema/spacewalk/common/tables/rhnPackageChangeLogData.sql]: Good point, thanks.

File: schema/spacewalk/common/tables/rhnPackageChangeLogData.sql [hunk 4]
```diff
@@ -13,6 +13,7 @@
 -- in this software or its documentation.
 --
 
+CREATE EXTENSION IF NOT EXISTS pgcrypto;
```
Comment (mackdk) [inline schema/spacewalk/common/tables/rhnPackageChangeLogData.sql:16]: I feel this should go in `schema/spacewalk/postgres/start.sql`. I see we have there already something for dblink there but I'm not sure to understand how that works. Maybe @mcalmer has more clues.
